api/src/pcapi/core/offers/api.py

```diff
@@ -992,6 +1004,56 @@ def set_offer_status_based_on_fraud_criteria(
     return status
 
 
+def resolve_offer_validation_sub_rule(
+    sub_rule: models.OfferValidationSubRule,
+    offer: AnyOffer,
+) -> bool:
+    if sub_rule.model:
+        if sub_rule.model.value in OFFER_LIKE_MODELS and type(offer).__name__ == sub_rule.model.value:
+            object_to_compare = offer
+        elif sub_rule.model.value == "CollectiveStock" and isinstance(offer, educational_models.CollectiveOffer):
+            object_to_compare = offer.collectiveStock
+        elif sub_rule.model.value == "Venue":
+            object_to_compare = offer.venue
+        elif sub_rule.model.value == "Offerer":
+            object_to_compare = offer.venue.managingOfferer
+        else:
+            raise exceptions.UnapplicableModel()
+
+        target_attribute = getattr(object_to_compare, sub_rule.attribute.value)
+    else:
+        target_attribute = type(offer).__name__
+
+    return OPERATIONS[sub_rule.operator.value](target_attribute, sub_rule.comparated["comparated"])  # type: ignore[operator]
+
+
+def resolve_offer_validation_rule(
+    rule: models.OfferValidationRule,
+    offer: AnyOffer,
+) -> bool:
+    is_offer_flagged = all(resolve_offer_validation_sub_rule(sub_rule, offer) for sub_rule in rule.subRules)
+    return is_offer_flagged
+
+
+def set_offer_status_based_on_fraud_criteria_v2(
+    offer: AnyOffer,
+) -> models.OfferValidationStatus:
+    offer_validation_rules = models.OfferValidationRule.query.all()
+
+    flagging_rules = dict()
+    for rule in offer_validation_rules:
+        if resolve_offer_validation_rule(rule, offer):
+            flagging_rules[rule.id] = rule.name
```

Comment:
(micro-détail / question) Pourquoi stocker les règles qui "matchent" dans un dictionnaire ? J'ai l'impression qu'on veut seulement savoir s'il y en a, mais qu'on ne se sert pas de leur valeur.

```suggestion
    flagging_rules = set()
    for rule in offer_validation_rules:
        if resolve_offer_validation_rule(rule, offer):
            flagging_rules.add(rule.id)  # ou `rule.id` si `rule` n'est pas _hashable_
```

D'autre part, est-ce qu'on ne pourrait pas s'arrêter à la première règle qui matche ?

```suggestion
    flagged = False
    for rule in offer_validation_rules:
        if resolve_offer_validation_rule(rule, offer):
            flagged = True
            break
    if flagged:
        status = models.OfferValidationStatus.PENDING
    else:
        status = models.OfferValidationStatus.APPROVED
```

Reply:
C'est en prévision !
Le métier a émis le besoin de savoir quelles règles causaient le flag d'une offre, ce qui est une info à rajouter à une table qui n'est pas encore prévu dans ce ticket. Mais je garde malgré tout l'info pour l'instant.